Overlay controlled (#3125)
jpmorganchase/salt-ds: .changeset/plenty-schools-fix.md

---
"@salt-ds/lab": minor
---

> Removed `onClose` prop from `Overlay`, onOpenChange is called for events that open/close the overlay.

Reviewer: ```suggestion
Removed the `onClose` prop from `Overlay`, `onOpenChange` is now called for events that open/close the overlay.
```

Author: Will fix in a follow up pr

---
"@salt-ds/lab": minor
---

Removed `onClose` prop from `Overlay`, onOpenChange is called for events that open/close the overlay.

```tsx
export const ControlledOverlay = () => {
  const [open, setOpen] = useState(false);
  const onOpenChange = (newOpen: boolean) => setOpen(newOpen);

  return (
    <Overlay open={open} onOpenChange={onOpenChange}>
      <OverlayTrigger>
        <Button

>           onClick={() => {
>             setShow(true);
>           }}

Reviewer: ```suggestion
```

---
"@salt-ds/lab": minor
---

> Remove `onClose` prop from `Overlay`, onOpenChange prop required in controlled state

Reviewer 2: Does this mean uncontrolled can't be used?

Author: The overlay manages the open state by default in an uncontrolled state - if the user wants something additional to happen when the close button is clicked then yes the overlay would have to be in a controlled state

Reviewer: We can remove this:

```suggestion
Removed `onClose` prop from `Overlay`. `onOpenChange` is called for events that open/close the overlay.
```
